### ISSUE TITLE
Resolve "Not a valid bip39 mnemonic" for Trezor recovery phrase entry

### ISSUE THREAD

**Issue by markmhx** [blockstack/stacks-wallet#584]
A user has reported that they see a `Not a valid bip39 mnemonic` error message when entering their 24-word Trezor recovery phrase in software mode with v4.0.4 that was previously used with the v3.x wallet.

I've inquired about the operating system they're using. This seems possibly related to the change made in https://github.com/blockstack/stacks-wallet/issues/239.

**Comment by markmhx** [comment]
@kyranjamie @timstackblock Is there any additional information from the user you think might be helpful in diagnosing this behavior?

**Comment by kyranjamie** [comment]
I'm not sure what to recommend. The input mnemonic is validated against bip39. 

I'd recommend the user trying to enter their seed phrase with the following formatting, incase it's the formatting that's breaking:
- All lowercase
- Separate with only a space
- No preceding numbers or other indication marks

**Comment by aseoconnor** [comment]
I'm the user who reported the issue. I'm using a brand new MacBook pro (m1 chip) on macOS 11.0.1.

I've validated that this key phrase is correct with my trezor. I also ordered a new ledger to see if I can transfer my stacks there and avoid this issue.

**Comment by kyranjamie** [comment]
Thanks @aseoconnor, let us know how it goes. This is not a particularly easy one to debug.

**Comment by fluidvoice** [comment]
There is one user on Telegram claiming same error. They are just using the Windows wallet without any hardware wallet.
They say the seed phrase works on v3.1.1 but not on any of the v4 releases.

**Comment by jeffdomke** [comment]
Can we test this scenario just to confirm we can't reproduce it? Also, one permutation that could get to this state is if they are using the 25th word as well?

**Comment by markmhx** [comment]
@aseoconnor Mind double checking that the phrase you tried entering from your Trezor has exactly 12 or 24 words?

**Comment by markmhx** [comment]
> There is one user on Telegram claiming same error. They are just using the Windows wallet without any hardware wallet.
> They say the seed phrase works on v3.1.1 but not on any of the v4 releases.
> https://t.me/BlockstackChat/224684

I've asked this user to email me so we can confirm if it's the same behavior initially reported with this issue.

**Comment by fluidvoice** [comment]
> > There is one user on Telegram claiming same error. They are just using the Windows wallet without any hardware wallet.
> > They say the seed phrase works on v3.1.1 but not on any of the v4 releases.
> > https://t.me/BlockstackChat/224684
> 
> I've asked this user to email me so we can confirm if it's the same behavior initially reported with this issue.

@markmhx This problem was reported on Telegram again with v4.1 wallet (no trezor) w/same error "Not a valid bip39 mnemonic"
user typed the seed into a text editor first then copy-pasted it into the wallet and it then worked.
telegram thread >> https://t.me/BlockstackChat/229595

**Comment by markmhx** [comment]
> Typing the seed into a text editor first then copy-pasting into the wallet worked.

Hmm might that suggest an issue with whitespace, which somehow the text editor copy/paste step resolves? 🤔

**Comment by markmhx** [comment]
I'm moving into the backlog and assigning @timstackblock so he can attempt to recreate soon.

**Comment by markmhx** [comment]
Note for anyone who encounters this error message:

Please double check that all the exact words you've entered are found in the following list and doesn't consist of any others: https://github.com/bitcoin/bips/blob/master/bip-0039/english.txt

**Comment by markmhx** [comment]
@kyranjamie Are non-English words compatible as well as long as they're provided here for BIP-39? https://github.com/bitcoin/bips/tree/master/bip-0039

**Comment by markmhx** [comment]
I'm modifying this issue to add a learn more link to the BIP-39 error message, which goes to:

https://www.hiro.so/questions/what-are-secret-keys-and-how-do-they-work#Secret%20Keys%20must%20be%20BIP-39%20compatible

Let's update the error copy in general as well while we're at it:

> Your Secret Key entry is not BIP-39 compatible. [Learn more](https://www.hiro.so/questions/what-are-secret-keys-and-how-do-they-work#Secret%20Keys%20must%20be%20BIP-39%20compatible)

![image](https://user-images.githubusercontent.com/28991/109653630-8a768780-7b61-11eb-9b8d-f0fb98b94172.png)

**Comment by kyranjamie** [comment]
> Are non-English words compatible as well as long as they're provided here for BIP-39?
@markmhx No.

I wasn't sure though. Also, I noticed this in the bip39 readme, https://github.com/bitcoinjs/bip39#reminder-for-developers

Ref: https://iancoleman.io/bip39/#spanish

**Comment by markmhx** [comment]
Would it be easier to add all language word lists or the "warning and proceed" approach recommended by that first link?

**Comment by timstackblock** [comment]
This worked for me without an issue, just typed everything in and it opened the wallet. I did get an error the first time but it was because I missed a word

video of me doing this:
http://somup.com/crenoJDJxW

**Comment by kyranjamie** [comment]
> Would it be easier to add all language word lists or the "warning and proceed" approach recommended by that first link?

Has anyone asked for this? It would complicate things, and I'm not sure why we'd do it when we generate english words.

**Comment by markmhx** [comment]
@kyranjamie Nope, no requests yet. If it's not an easy change, I agree we can wait until someone says they need non-English entry for some reason.

**Comment by ghostshirt** [comment]
I can confirm this bug for both on Ubuntu and Windows 4.1.0 wallets. I have checked all the words against BIP-0039 English words. I have copied from a plain text file with only single white spaced between them.

**Comment by kyranjamie** [comment]
@ghostshirt Can you clarify where the seed came from? Is it also the same phrase as from a Trezor device?

**Comment by ghostshirt** [comment]
@kyranjamie Seed came from an earlier wallet long before 2.0, mainnet STX on it. 24-word seed. I've just downloaded the newest wallets and tested them to no avail. I'll try with 3.0 version wallet on a Windows box later.

**Comment by kyranjamie** [comment]
@ghostshirt This is a tricky one to debug. As far as I'm aware, we've never generated non-bip39 compliant seeds.

Do you have the [STX CLI installed](https://docs.blockstack.org/references/stacks-cli)? This doesn't validate for bip39 compliance. You can derive your address from your seed using `stx make_keychain "your seed phrase"` command. Does this result in the address you expect?

**Comment by ghostshirt** [comment]
@kyranjamie Yes, this works (`stx make_keychain` command).

**Comment by kyranjamie** [comment]
Thanks @ghostshirt, that's really helpful to know.

@yknl any insights here? Did the old wallet previously not generate bip39 compliant seeds? Did it accept seeds for _restore_ without checking for bip39 compliance?

**Comment by yknl** [comment]
The wallet has generated bip39 seed phrases since v1.0
See https://github.com/blockstack/stacks-wallet/blob/v1.0.0/app/actions/wallet.js#L84
